### PR TITLE
Fix issues #280: LIST "start or end error" after deleting the last item

### LINKED ISSUE #280: LIST "start or end error" after deleting the last item from list
**Describe the bug**
It seems that `reading` the list breaks, after deleting the last item from it.

**To Reproduce**
Steps to reproduce the behavior(Be specific!):
1.   Create a list and push one item
```
err = tx.RPush("userItemIds", []byte("user1"), []byte("item1"))
if err != nil {
	return fmt.Errorf("failed to push ids list: %v", err)
}
```
2.  Remove the item
```
_, err = tx.LRem("userItemIds", []byte("user1"), 0, []byte("item1"))
if err != nil {
	return err
}
```

3.  Retrieve the list
```
err := q.DB.View(func(tx *nutsdb.Tx) error {
data, err := tx.LRange("userItemIds", []byte("user1"), 0, -1)
if errors.Is(err, nutsdb.ErrBucket) || errors.Is(err, list.ErrListNotFound) {
	return nil
}

....

return nil
})
```

**Expected behavior**
The item is removed, retrieving the list throws "list.ErrListNotFound"

**What actually happens**
Retrieving the list throws: `start or end error`.

**please complete the following information :**
 - OS: [Ubuntu 22]
 - NutsDB Version [v0.11.1]

**Additional context**
The list can't be read after trying to delete the last item. it produces the same error ``start or end error`.

### PR DESCRIPTION
**Pull request by ShawnHXH**
Fix a bug reported in Issues #280

### CODE REVIEW

File: ds/list/list.go [hunk 1]
```diff
@@ -173,11 +173,8 @@ func (l *List) Size(key string) (int, error) {
 // LRange returns the specified elements of the list stored at key
 // [start,end]
 func (l *List) LRange(key string, start, end int) (list [][]byte, err error) {
-	if l.IsExpire(key) {
-		return nil, ErrListNotFound
-	}
 	size, err := l.Size(key)
-	if err != nil {
+	if err != nil || size == 0 {
```
Comment (elliotchenzichang) [inline ds/list/list.go:177]: Maybe warping this logic as a function is a better practice? Like l.empty()? WDYT?

Comment (ShawnHXH) [reply ds/list/list.go:177]: Sure, it will be more logical.

Comment (elliotchenzichang) [reply ds/list/list.go:177]: @ShawnHXH Yep, I think your idea is great.

Comment (ShawnHXH) [reply ds/list/list.go:177]: Sounds great

**Review by elliotchenzichang — COMMENTED** [review]
LGTM

**Pull request merged**